### PR TITLE
Added role value in the user display info

### PR DESCRIPTION
**Pull request by martinmitrevski**
### 🔗 Issue Link
Resolves https://github.com/GetStream/stream-chat-swiftui/issues/458.

### 🎯 Goal

Extend the user display info to contain the user's role.

### 🛠 Implementation

_Provide a description of the implementation._

### 🧪 Testing

_Describe the steps how this change can be tested (or why it can't be tested)._

### 🎨 Changes

_Add relevant screenshots or videos showcasing the changes._

### ☑️ Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created)

### CODE REVIEW

File: StreamChatSwiftUITests/Tests/Utils/ViewFactory_Tests.swift [hunk 1]
```diff
@@ -123,7 +123,12 @@ class ViewFactory_Tests: StreamChatTestCase {
     func test_viewFactory_makeMessageAvatarView() {
         // Given
         let viewFactory = DefaultViewFactory.shared
-        let userInfo = UserDisplayInfo(id: .unique, name: .unique, imageURL: URL(string: "https://example.com"))
+        let userInfo = UserDisplayInfo(
+            id: .unique,
+            name: .unique,
+            imageURL: URL(string: "https://example.com"),
+            role: .user
```
Comment (testableapple) [inline StreamChatSwiftUITests/Tests/Utils/ViewFactory_Tests.swift:130]: Do we need the tests for different roles?

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 # Upcoming
 
+### ✅ Added
+- Role value in the user display info
```
Comment (testableapple) [inline CHANGELOG.md:7]: Does it make sense to add something to the DemoApp?

**Review by testableapple — APPROVED** [review]
LGTM, just a couple of clarifying questions 👍

File: CHANGELOG.md (hunk 2 above)
Comment (martinmitrevski) [reply CHANGELOG.md:7]: Customers would use it to show role badge in the avatar, but we don't have such design for the demo app. Not sure if we can use it elsewhere? Open to ideas.

Comment (testableapple) [reply CHANGELOG.md:7]: Was thinking about adding something in the channel's info besides users' names, but now not sure it's worth it, to be honest. Let's not overcomplicate this 👍

**Pull request merged**